gourmet/backends/db.py

```diff
@@ -1539,7 +1539,7 @@ def sort_groups (x,y):
             elif group_order[x[0]] == group_order[y[0]]: return 0
             else: return -1
         alist=list(groups.items())
-        alist.sort(sort_groups)
+        sorted(alist, key=sort_groups)
```

Comment:
This doesn't sort in place. `sorted` creates a generator with the sorted elements.

Reply:
fine would be `alist = list(sorted(groups.items(), key=sort_groups), but new-style key-functions are not comparators. You have to replace this with a key function (see #9), and if you don't want to write one, you can use functools to create one (https://docs.python.org/3/library/functools.html#functools.cmp_to_key)

Reply:
I did not know about `cmp_to_key`, thanks!